js/gestionPresupuestoWeb.js

```diff
@@ -1,60 +1,201 @@
-        </div>
-        `;
+        let divAgrupacion = document.createElement("div");
+        divAgrupacion.className = "agrupacion-dato";
+
+        let spanClave = document.createElement("span");
+        spanClave.className = "agrupacion-dato-clave";
+        spanClave.textContent = `${key}`;
+
+        let spanValor = document.createElement("span");
+        spanValor.className = "agrupacion-dato-valor";
+        spanValor.textContent = `${value}`;
+
+        divAgrupacion.append(spanClave);
+        divAgrupacion.append(spanValor);
+
+        div.append(divAgrupacion);
     }
 
-    mostrar.innerHTML = `
-    <div class="agrupacion">
-        <h1>Gastos agrupados por ${periodo}</h1>
-        ${txtEtiquetas}
-    </div>
-`;
+    mostrar.append(div);
+
+}
+
+function repintar(){
+    let presupuesto = gp.mostrarPresupuesto();
+    mostrarDatoEnId("presupuesto", presupuesto);
+
+    let gTotales = gp.calcularTotalGastos();
+    mostrarDatoEnId("gastos-totales", gTotales);
+
+    let bTotal = gp.calcularBalance();
+    mostrarDatoEnId("balance-total", bTotal);
+
+    document.getElementById("listado-gastos-completo").innerHTML = '';
+
+    let lgastos = gp.listarGastos();
+
+    for(let gasto of lgastos){
+        mostrarGastoWeb("listado-gastos-completo", gasto);
+    }
+}
 
+function actualizarPresupuestoWeb(){
+    let presupuesto = prompt('Introduce un presupuesto nuevo');
+    presupuesto = parseInt(presupuesto);
+
+    gp.actualizarPresupuesto(presupuesto);
+
+    repintar();
+}
+
+function nuevoGastoWeb(){
+    let descripcion = prompt('Introduce la descripción del gasto');
+    let valor = prompt('Introduce el valor del gasto');
+    let fecha = prompt('Introduce la fecha del gasto');
+    let etiquetas = prompt('Introduce las etiquetas');
+
+    valor = parseFloat(valor);
+    etiquetas = etiquetas.split(',');
+
+    let gasto1 = new gp.CrearGasto(descripcion, valor, fecha, etiquetas);
+
+    gp.anyadirGasto(gasto1);
+
+    repintar();
+}
+
+function EditarHandle(){
+
+    this.handleEvent = function(e){
+       //Pedir datos al usuario
+       let desc = prompt("Introduce la descripción nueva", this.gasto.descripcion);
+       let val = prompt("Introduce el valor nuevo", this.gasto.valor);
+       let fech = prompt("Introduce la fecha nueva", this.gasto.fecha);
+       let etiq = prompt("Inroduce las etiquetas nuevas", this.gasto.etiquetas);
+
+       val = parseFloat(val);
+       etiq = etiq.split(',');
+
+       this.gasto.actualizarDescripcion(desc);
+       this.gasto.actualizarValor(val);
+       this.gasto.actualizarFecha(fech);
+       this.gasto.anyadirEtiquetas(...etiq);
+
+       repintar();
+    }
+}
+
+function BorrarHandle(){
+    this.handleEvent = function(e){
+
+        gp.borrarGasto(this.gasto.id);
+
+        repintar();
+    }
+}
+
+function BorrarEtiquetasHandle(){
+    this.handleEvent = function(e){
+
+        this.etiqueta = this.etiqueta.split(',');
+
+        this.gasto.borrarEtiquetas(...this.etiqueta);
```

Comment:
No hace falta. `borrarEtiquetas` admite lista de parámetros. Y `this.etiqueta` es un string. Por tanto, puedes hacer `this.gasto.borrarEtiquetas(this.etiqueta)`.

Reply:
Cierto. Muchas gracias :)